src/storage-operators/src/persist_source.rs

```diff
@@ -365,6 +365,13 @@ impl PendingWork {
             match (key, val) {
                 (Ok(SourceData(Ok(row))), Ok(())) => {
                     if let Some(mfp) = map_filter_project {
+                        // We originally accounted work as the number of outputs, to give downstream
+                        // operators a chance to reduce down anything we've emitted. This mfp call
+                        // might have a restrictive filter, which would have been counted as no
+                        // work. However, in practice, we've been decode_and_mfp be a source of
+                        // interactivity loss during rehydration, so we now also count each mfp
+                        // evaluation against our fuel.
+                        *work += 1;
```

Comment:
wondering if we want to move this up even further... both `FetchedPart::next` and `until.less_equal(&time)` can filter out an arbitrary number of tuples before producing a candidate for mfp to evaluate

Reply:
Good point!

Not sure what we could do about the filtering inside `FetchedPart::next` but maybe that's okay because it's only looking at the timestamp (we always return after decoding key and val, which is presumably the expensive part). 

The `until.less_equal(&time)` is probably not a terribly selective filter in the rehydration case, but might be worth it in the spirit of being defensive. We'd probably be double counting work then in the common case, no idea if we would feel bad about that. Maybe we should change it to be "1 work per decoded key/val` plus `mfp_output.len().saturating_sub(2)` (the latter basically being how many "extra" outputs are generated)

Reply:
Double counting seems fine to me, it's all an approximation right now. I think I'd feel better about that if the yield threshold were configurable

Reply:
other thought: we could add a `debug!` that outputs the time spent before yielding and # of inputs and # of outputs (or create additional metrics for those) to guide any future tuning

Reply:
Given that there's a decent amount of hesitation around this, I've decided to keep the change as minimal as possible. Adding something for each mfp eval is pretty critical, so I think we have to live with that change, but I think moving the filtering is a bit speculative.

If possible, I'd also love to punt on any additional metrics or introspection in this PR because I don't really have a ton of time to devote to it atm, but I really think we want the knob in prod.